### PR TITLE
Keyboard: add support for kbd19x

### PR DESCRIPTION
**Pull request by jshuf**
Add support for the kbd19x kit from KBDfans.
* The keyboards/kbd19x and keyboards/kdb19x/keymaps/default readme.md document the board and the default layout, respectively.
* Basic backlight and RGB underglow support have been enabled.
* The default keymap has been tested against a fresh kbd19x build.

### CODE REVIEW

File: keyboards/kbd19x/kbd19x.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  {k10, k11, k12, k13, k14, k19, k1a, k1b, k1c, k1f, k1g, k1h, k1i},\
+  {k20, k21, k22, k23, k24, k29, k2a, k2b, k2c, k2f, k2g, k2h, k2i},\
+  {k30, k32, k33, k34, k35, k3a, k3b, k3c, k3e, k3f, k3g, k3h, k3i},\
+  {k40, k41, k42, k43, k44, k49, k4a, k4b, k4c, k4f, k4g, k4h, k4i},\
+  {k50, k51, k52, k5e, k5f, k59, k5a, k5b, k5d, k56, k5g, k5h, k5i},\
+  {k05, k06, k07, k08, k15, k16, k17, k18, k25, k26, k27, k28, k36},\
+  {k1d, k1e, k0e, k2e, k4e, k37, k38, k39, k45, k46, k47, k48, XXX},\
+}
+
+#define LAYOUT_default( \
```
Comment (drashna) [inline keyboards/kbd19x/kbd19x.h]: @noroadsleft and/or @mechmerlin was there a better name for this layout that we wanted to use for the community layouts?

Comment (noroadsleft) [reply keyboards/kbd19x/kbd19x.h]: @drashna Not for community layouts, I don't think. The KBD19x has a bit of an unusual layout.

https://kbdfans.cn/products/pre-orderkbdfans-kbd19x-90-mechanical-keyboard-diy-kit

It's not quite 96-key / compact 1800.

If anything, I'd say rename `LAYOUT_default` as `LAYOUT_ansi`. @jshuf already included the matrix positions needed to change layouts (thanks for that!) in the comments above the macros.

File: keyboards/kbd19x/info.json [hunk 2]
```diff
@@ -0,0 +1,219 @@
+{
+    "keyboard_name": "KBD19x", 
+    "url": "https://github.com/qmk/qmk_firmware/tree/master/keyboards/kbd19x", 
+    "maintainer": "qmk", 
+    "bootloader": "",
```
Comment (mechmerlin) [inline keyboards/kbd19x/info.json]: This line is not needed

```suggestion

```

File: keyboards/kbd19x/rules.mk [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (mechmerlin) [inline keyboards/kbd19x/rules.mk]: We can remove this line
```suggestion

```

File: keyboards/kbd19x/kbd19x.h (hunk 1 above)
Comment (jshuf) [reply keyboards/kbd19x/kbd19x.h]: Agreed renaming to ansi makes sense.  Change https://github.com/qmk/qmk_firmware/pull/4272/commits/9913c7b2def48cb67858ea7d7ca88c37c127f352 added to the PR.

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**